index.js

```diff
@@ -22,7 +22,16 @@ class rpRetry {
                     logger.info(`Encountered error ${err.message} for ${options.method} request to ${options.uri}, retry count ${tryCount}`);
                     tryCount -= 1;
                     if (tryCount) {
-                        return fetchDataWithRetry(tryCount);
+                        let delay = options.delay || 100; // default delay between retries
+                        if (options.factor) {
+                            delay *= options.factor;
+                        }
+                        return new Promise((resolve, reject) => {
+                            setTimeout(() => {
+                                logger.debug(`waiting for ${delay} ms before next retry for ${options.uri}`);
+                                resolve(fetchDataWithRetry(tryCount));
+                            }, delay);
+                        });
```

Comment:
Consider multiplying the delay by the  factor *after* the fetch. 

On might expect the second request to happen 100 ms after the first request, but currently it's 100 ms * factor instead.

Reply:
@markstos nope, this is in the `catch` clause. So as I understand it, this is already on the first failure. I did have the same thought as you are, and agree waiting on the first download would be wasteful :-).